### PR TITLE
Actiation functions new template

### PR DESCRIPTION
**Pull request by kmintoff**
'Fixed' overflowing citations by putting them on a new line. This will need to be reviewed if bibliography style is changed later on.

### CODE REVIEW

**Comment by kmintoff** [comment]
Closing PR to update with new citation style

File: terms/activationfunctions_km.bib [hunk 1]
```diff
@@ -0,0 +1,103 @@
+@BOOK{Snyman2005,
```
Comment (jp-um) [inline terms/activationfunctions_km.bib]: From all entries remove language and doi

File: terms/activationfunctions_km.tex [hunk 2]
```diff
@@ -0,0 +1,114 @@
+\chapter[Activation Functions]{Activation Functions}
+\label{ch:activation-functions}\index{activation functions|(}
+``Neural networks were originally conceived as a model that would imitate the function of the human brain---a set of neurons joined together by a set of connections. Neurons, in this context, are composed of a weighted sum of their inputs followed by a nonlinear function, which is also known as an activation function.'' \cite{caterini2018}
```
Comment (jp-um) [inline terms/activationfunctions_km.tex]: Should open a little bit softer "Del Piero et al. state that ...."

File: terms/activationfunctions_km.tex [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+Activation Functions are used in Artificial Neural Networks to determine whether the output of the neuron should be considered further or ignored. If the activation function chooses to continue considering the output of a neuron, we say that the neuron has been activated. The output of the activation function is what is passed on to the subsequent layer in a multilayer neural network. To determine whether a neuron should be activated, the activation function takes the output of a neuron and transforms it into a value commonly bound to a specific range, typically from $0$ to $1$ or $-1$ to $1$ depending on the which activation function is applied.
```
Comment (jp-um) [inline terms/activationfunctions_km.tex]: No need for maths mode when you have 0s or 1s,  Please.

File: terms/activationfunctions_km.tex [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+\begin{equation}\label{stepfunctionderivative}
+    \frac{d}{d(x)}f(x) =
+      \begin{dcases*}
+                                       0 & \text{for \(x \neq 0\)} \\
+                                       ? & \text{for \(x = 0\)} \\
+      \end{dcases*}
+\end{equation}
+
+The Heavside step function, also known as binary step function, is one of the simplest activation functions that can be used in a neural network. This activation function returns $0$ if the input of a node is less than a predetermined threshold (typically $0$), or otherwise it returns $1$ if the output of the node is greater than or equal to the threshold. This function was first used in a machine learning context in 1957 by Frank Rosenblatt in his seminal work describing the perceptron, the precursor to the modern day neural network \cite{rosenblatt1957perceptron}.
```
Comment (jp-um) [inline terms/activationfunctions_km.tex]: Which equation is this??  Label equations and refer them from text.

File: terms/activationfunctions_km.tex [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+\begin{equation}\label{sigmoidfunction}
+    f(x) = \frac{1}{(1 + e^{-x})}
+\end{equation}
+
+\begin{equation}\label{sigmoidfunctionderivative}
+    \frac{d}{d(x)}f(x) = f(x)(1-f(x))
+\end{equation}
+
+The sigmoid function, also known as the logistic function, is one of the most commonly used activation functions in neural networks, because of its simplicity and desirable properties. The use of this function in neural networks was first introduced by Rummelhart, Hinton and Williams in one of the most important papers in the field of machine learning, which described the back-propagation algorithm and the introduction of hidden layers, giving rise to modern day neural networks \cite{DavidE.Rumelhart1986Lrbb}. \newline The values produced by the step function are bound between $0$ and $1$, both not inclusive, which help manage the exploding gradient problem. The function also has a very steep gradient for a relatively small range of values, typically in the range of $-2$ to $2$. This means that for most inputs that the function receives it will return values that are very close to either $0$ or $1$.
```
Comment (jp-um) [inline terms/activationfunctions_km.tex]: >  Rummelhart, Hinton and Williams

Use `\citet`.  No need for newlines (any longer)

File: terms/activationfunctions_km.tex [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    f(x) = \frac{(e^{x} - e^{-x})}{(e^{x} + e^{-x})}
+\end{equation}
+
+\begin{equation}\label{tanhfunctionderivative}
+    \frac{d}{d(x)}f(x) = 1-f(x)^2.
+\end{equation}
+
+The hyperbolic tangent ($tanh$) function is another activation function that is sometimes used instead of sigmoid. The $tanh$ function has the same characteristics of the sigmoid function mentioned above. In fact if one plots the $tanh$ function, one can observe that it is simply a scaled version of the sigmoid function. As a result of this scaling, the tanh function has a steeper gradient in towards the origin, and it returns values between $-1$ and $1$.
+
+Nowadays with the rise of deep learning, these functions are becoming less commonly used. Xavier Glorot and Yoshua Bengio studied in detail the effects of the sigmoid and tanh activation functions. They note how the sigmoid function in particular is not well suited for deep networks with random initialization \cite{glorot2010understanding}.
```
Comment (jp-um) [inline terms/activationfunctions_km.tex]: should not hardcode reference EVER.  use `\citet`

File: terms/activationfunctions_km.tex [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+\section{Rectified Linear Unit}\label{sec:relu}\index{rectified linear unit (relu)}
+\begin{marginfigure}
```
Comment (jp-um) [inline terms/activationfunctions_km.tex]: Figure should be referred to in text via labels.

File: terms/activationfunctions_km.tex [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+                                       1 & \text{for $x \geq 0$} \\
+      \end{dcases*}
+\end{equation}
+
+The ReLU function returns $0$ if the input of the function is negative, otherwise it outputs the value of the input itself. This function is non-linear in nature even though at first glance it may seem similar to an identity function. The ReLU function is becoming one of the more commonly used activation function due to its simplicity performance, and suitability to networks with many layers. Another benefit of the ReLU function is that it produces sparse activations (not all nodes in the network are activated) unlike the sigmoid or hyperbolic tangent functions. 
+
+The ReLU function has been used in many neural network models to improve their performance. Naid and Hinton use ReLU to improve the performance of Restricted Boltzmann Machines in object recognition \cite{Nair2010}. In 2012, a breakthrough Convolutional Neural Network (CNN) architecture called AlexNet pioneered the use of the ReLU activation function together with dropout layers to minimise over fitting in CNNs. \newline \cite{KrizhevskyAlex2017Icwd}.
```
Comment (jp-um) [inline terms/activationfunctions_km.tex]: no newlines, again dont hard-code refs and authors.

**Pull request closed without merging**